backend/alembic/versions/0024_sibling_roms_db_view.py

```diff
@@ -0,0 +1,68 @@
+"""empty message
+
+Revision ID: 0024_sibling_roms_db_view
+Revises: 0023_make_columns_non_nullable
+Create Date: 2024-08-08 12:00:00.000000
+
+"""
+
+import sqlalchemy as sa
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "0024_sibling_roms_db_view"
+down_revision = "0023_make_columns_non_nullable"
+branch_labels = None
+depends_on = None
+
+
+def upgrade() -> None:
+    with op.batch_alter_table("roms", schema=None) as batch_op:
+        batch_op.create_index("idx_roms_igdb_id", ["igdb_id"])
+        batch_op.create_index("idx_roms_moby_id", ["moby_id"])
+
+    connection = op.get_bind()
+
+    connection.execute(
+        sa.text(
+            """
+            CREATE VIEW sibling_roms AS
+            SELECT
+                r1.id AS rom_id,
+                r2.id AS sibling_rom_id,
+                r1.platform_id AS platform_id,
+                COALESCE(r1.igdb_id, r2.igdb_id) AS igdb_id,
+                COALESCE(r1.moby_id, r2.moby_id) AS moby_id,
```

Comment:
A few considerations we can discuss here:

* If we add more metadata providers, we would need to add its sibling logic here. Adding it to the `WHERE` is easy, but adding the new ID field here could require a DROP + CREATE if the field is not appended at the end of the current SELECT. Maybe we can move `created_at` and `updated_at` either at the beginning of the list of fields, or right after `sibling_rom_id`, so future migrations maintain similar fields close by.

```
rom_id
sibling_rom_id
created_at
updated_at
platform_id
igdb_id
moby_id
# New metadata ID fields...
```

* The current implementation using `COALESCE` could be confusing, if sibling roms match in one of the IDs but not the other. For example, if the `igdb_id` matches but `moby_id` does not, then the field value is not clear (it could be `r1.moby_id` if it's populated, even if `r2.moby_id` is different, or `r2.moby_id` if r1 has the field empty). I think the solution for this is to either avoid these fields, or only populate them when they match:

```sql
CASE WHEN r1.igdb_id <=> r2.igdb_id THEN r1.igdb_id AS igdb_id
```

Reply:
Aight solid! If you see any other ways it can be improved push them directly to the branch.